packages/typescript-plugin/src/namedPipe/server.ts

```diff
@@ -0,0 +1,62 @@
+import * as fs from 'fs';
+import * as net from 'net';
+import { collectExtractProps } from './requests/collectExtractProps';
+import { getPropertiesAtLocation } from './requests/getPropertiesAtLocation';
+import { getQuickInfoAtPosition } from './requests/getQuickInfoAtPosition';
+import { pipeFile } from './utils';
+
+export interface CollectExtractPropsRequest {
+	type: 'collectExtractProps';
+	fileName: string;
+	templateCodeRange: [number, number];
+}
+
+export interface GetPropertiesAtLocationRequest {
+	type: 'getPropertiesAtLocation';
+	fileName: string;
+	position: number;
+}
+
+export interface GetQuickInfoAtPosition {
+	type: 'getQuickInfoAtPosition';
+	fileName: string;
+	position: number;
+}
+
+export type Request = CollectExtractPropsRequest
+	| GetPropertiesAtLocationRequest
+	| GetQuickInfoAtPosition;
+
+let started = false;
+
+export function startNamedPipeServer() {
+	if (started) return;
+	started = true;
+	const server = net.createServer(connection => {
```

Comment:
Is it not possible to communicate using node-ipc for example? At least that's what typescript-language-server is using to communicate with tsserver but not sure if that's relevant for this case...

Reply:
In my memory IPC communication requires obtaining a ChildProcess instance? Neither the LSP server nor the language client can obtain a process instance of tsserver so this may not be possible.